### PR TITLE
Enhancement/sleep schedule check if current sleep is over

### PR DESCRIPTION
**Pull request by alexyaoyang**
## Fixes (provide links to github issues if you can):
- Currently, if we set up the sleep schedule task such that the bot should be sleeping now and start the bot, the bot will not sleep and continue to run until next sleep cycle before starting to sleep.
- Changed it to start sleeping whenever it should be sleeping, no matter when the bot is started.
- Added config field "skip_current_sleep_cycle" (defaults to true)

### CODE REVIEW

File: pokemongo_bot/cell_workers/sleep_schedule.py [hunk 1]
```diff
@@ -60,6 +61,8 @@ def _process_config(self):
             timedelta(
                 hours=duration_random_offset.hour, minutes=duration_random_offset.minute).total_seconds())
 
+        self.skip_current_sleep_cycle = self.config.get('skip_current_sleep_cycle', True)
```
Comment (Quantra) [inline pokemongo_bot/cell_workers/sleep_schedule.py]: This shouldn't change during `work()` set it in `initialize()`

**Comment by Quantra** [comment]
I tried and failed to come up with a neat solution for this.  Yours looks good at first glance =]

File: pokemongo_bot/cell_workers/sleep_schedule.py (hunk 1 above)
Comment (alexyaoyang) [inline pokemongo_bot/cell_workers/sleep_schedule.py]: It is in `_process_config()`, not `work()` :)

**Comment by alexyaoyang** [comment]
@Quantra Thanks 👍

File: pokemongo_bot/cell_workers/sleep_schedule.py (hunk 1 above)
Comment (Quantra) [inline pokemongo_bot/cell_workers/sleep_schedule.py]: doh! skimmed over the `@@ -60,6 +61,8 @@ def _process_config(self):`  Sorry!  Carry on ;)

**Comment by TheSavior** [comment]
If you are adding `skip_current_sleep_cycle` so that you can support not changing behavior, I really appreciate that. However, I think in this case the current behavior isn't expected by people so I'd actually be fine with not having that config and always sleep when it is meant to sleep.

Up to you though, if you think we should have the config, then I'm totally on board with that.

**Comment by alexyaoyang** [comment]
@TheSavior Yes, I was thinking of the user testing the bot functions before enabling the SleepSchedule but `enabled` will be sufficient. Removed `skip_current_sleep_cycle ` from this PR.

**Comment by TheSavior** [comment]
👍

**Pull request merged**